public/app/features/migrate-to-cloud/onprem/Page.tsx

```diff
@@ -42,64 +44,93 @@ function useGetLatestMigrationDestination() {
   };
 }
 
-function useGetLatestMigrationRun(migrationUid?: string) {
-  const listResult = useGetCloudMigrationRunListQuery(migrationUid ? { uid: migrationUid } : skipToken);
-  const latestMigrationRun = listResult.data?.runs?.at(-1);
+const SHOULD_POLL_STATUSES: Array<SnapshotDto['status']> = [
+  'INITIALIZING',
+  'CREATING',
+  'UPLOADING',
+  'PENDING_PROCESSING',
+  'PROCESSING',
+];
 
-  const runResult = useGetCloudMigrationRunQuery(
-    latestMigrationRun?.uid && migrationUid ? { runUid: latestMigrationRun.uid } : skipToken
-  );
+const STATUS_POLL_INTERVAL = 5 * 1000;
+
+function useGetLatestSnapshot(sessionUid?: string) {
+  const [shouldPoll, setShouldPoll] = useState(false);
+
+  const listResult = useGetShapshotListQuery(sessionUid ? { uid: sessionUid } : skipToken);
+  const lastItem = listResult.data?.snapshots?.at(-1); // TODO: account for pagination and ensure we're truely getting the last one
+
+  const getSnapshotQueryArgs = sessionUid && lastItem?.uid ? { uid: sessionUid, snapshotUid: lastItem.uid } : skipToken;
+
+  const snapshotResult = useGetSnapshotQuery(getSnapshotQueryArgs, {
+    pollingInterval: shouldPoll ? STATUS_POLL_INTERVAL : 0,
+    skipPollingIfUnfocused: true,
+  });
+
+  useEffect(() => {
+    if (SHOULD_POLL_STATUSES.includes(snapshotResult.data?.status)) {
+      setShouldPoll(true);
+    } else {
+      setShouldPoll(false);
+    }
+  }, [snapshotResult?.data?.status]);
 
   return {
-    ...runResult,
+    ...snapshotResult,
 
-    data: runResult.data,
+    data: snapshotResult.data,
```

Comment:
I think this is redundant since `snapshotResult` already has the data attr?
```suggestion
```

Reply:
Yeah, good catch!